src/test/java/org/springframework/integration/aws/s3/config/xml/AmazonS3OutboundChannelAdapterParserTests.java

```diff
@@ -162,6 +161,22 @@ public void withHandlerAdviceChain() {
 		ctx.destroy();
 	}
 
+	/**
+	 * Test case for the xml definition with the Channel Attributes.
+	 */
+	@Test
+	public void withChannelAttributeImplementation() {
+		ClassPathXmlApplicationContext ctx = new ClassPathXmlApplicationContext("classpath:s3-valid-outbound-cases.xml");
+		EventDrivenConsumer consumer = ctx.getBean("withChannelAdapterAttributes", EventDrivenConsumer.class);
+		AmazonS3MessageHandler handler = getPropertyValue(consumer, "handler", AmazonS3MessageHandler.class);
+		assertEquals(DefaultAmazonS3Operations.class, getPropertyValue(handler, "operations").getClass());
+        assertEquals("input", getPropertyValue(consumer, "inputChannel.beanName", String.class));
+        assertEquals("US-ASCII", getPropertyValue(handler, "charset", String.class));
+		assertEquals(100, getPropertyValue(consumer, "phase", Integer.class).intValue());
+        assertEquals(true, getPropertyValue(consumer, "autoStartup", Boolean.class));
+		ctx.destroy();
```

Comment:
Thank you for contribution, @pckeyan !

Looks good, BTW!

I can fix all minor concerns on merge, but want to let you know anyway for your future contributions.

1. If you modify some class be sure do not change imports order, or change them according the Code Style rules: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Code-Style. Although we import in SI `static`s as first onces...
2. Be sure that you use tabs for intends. Not sure what is wrong in this your new test, but it looks freaky.
3. Please, use `ctx.close()` instead of `destroy()`.